pkg/workflow_execution.go

```diff
@@ -155,18 +162,30 @@ func (c *Client) injectAutomatedFields(namespace string, wf *wfv1.Workflow, opts
 			},
 		})
 
+		// Extend artifact credentials if only key is provided
+		if len(template.Outputs.Artifacts) > 0 {
+			for j, artifact := range template.Outputs.Artifacts {
```

Comment:
We can omit the `if len` check here, loop won't do anything if there are no items.